### ISSUE TITLE
Understanding the paper

### ISSUE THREAD

**Issue by debasishaimonk** [lifeiteng/valle#130]
<img width="671" alt="Screenshot 2023-05-31 at 6 29 45 PM" src="https://github.com/lifeiteng/vall-e/assets/115630737/d76403bf-fe81-4741-b7d9-23625a345929">

Could anyone please make me understand the difference between the first quantizer and acoustic prompt? As both of the Symbols looks to be same.

Atleast with respect to the code

**Comment by constan1** [comment]
The quantizer is the first layer Residual Quantizer (VQ-1) There are really two parts to this. 

1.  wavform is passed through the Encodec pretrained model to compress the wavform. ( about 75 frames/1 second of audio).  The residuals are computed for all 8 layers. This is used as the embedding space for the Transformer decoder model ( 12 layer 16 multi head attention layers). With an embedding space of 1024 ( number of codebook indices)

2. The transformer LM is trained within this embedding space in two parts ( AR and NAR). 

3. The resulting approximation Matrix is then sent through the Decoder of the Encodec model and outcomes the reconstructed wav form

In the equation above. A prompt segmentation of the original audio is used ( I think they mention 3 seconds, or 10 seconds. Don't remember. But they do segment it. ) This is used as the prompt or input to the Transformer Decoder. The rest of it is used as the target ( to compute loss). 

The equation above basically says "Given the phoneme sequence x, Prompt sequence C ( matrix as its time based), what is the probability of getting the target. The result is a response matrix which is compared against the target for loss.

**Comment by chenjiasheng** [comment]
"the first quantizer and acoustic prompt", they are different concept and thus can't be compared together.
I think you are talking about first quantize layer codes of the prompt audio and of the predicted continual audio.
As state in the paper, we do not distinguished these two segments when training AR, meaning there's no prompt at all, or you can say all preceding frames are prompt to the current frame.
At inference stage, given 3s or more initial audio segment, this 3s segment naturally becomes the prompt to all the future generated continual frames.
